diff --git a/applications/luci-app-vnstat/luasrc/model/cbi/vnstat.lua b/applications/luci-app-vnstat/luasrc/model/cbi/vnstat.lua
index c60c649..2b8aa65 100644
--- a/applications/luci-app-vnstat/luasrc/model/cbi/vnstat.lua
+++ b/applications/luci-app-vnstat/luasrc/model/cbi/vnstat.lua
@@ -63,13 +63,6 @@ function mon_ifaces.write(self, section, val)
 		end
 	end
 
-	for i, _ in pairs(ifaces) do
-		if not s[i] then
-			fs.unlink(dbdir .. "/" .. i)
-			fs.unlink(dbdir .. "/." .. i)
-		end
-	end
-
 	if next(s) then
 		m.uci:set_list("vnstat", section, "interface", utl.keys(s))
 	else
diff --git a/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua b/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua
index c0bb380..5ff0026 100644
--- a/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua
+++ b/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi.lua
@@ -249,6 +249,10 @@ if hwtype == "mac80211" then
 
 	s:taboption("advanced", Value, "frag", translate("Fragmentation Threshold"))
 	s:taboption("advanced", Value, "rts", translate("RTS/CTS Threshold"))
+
+	s:taboption("advanced", Flag, "noscan", translate("Force 40MHz"),
+	 translate("Do not scan for neighboring BSSes prior to enabling 40 MHz channel"))
+
 end
 
 
@@ -486,6 +490,12 @@ if hwtype == "mac80211" then
 	isolate:depends({mode="ap"})
 	isolate:depends({mode="ap-wds"})
 
+	disassoc = s:taboption("advanced", Flag, "disassoc_low_ack", translate("Disassociate stations with low ACK"),
+	 translate("Disassociate stations based on excessive transmission failures or other indications of connection loss"))
+	disassoc:depends({mode="ap"})
+	disassoc:depends({mode="ap-wds"})
+	disassoc.default = disassoc.enabled
+
 	ifname = s:taboption("advanced", Value, "ifname", translate("Interface name"), translate("Override default interface name"))
 	ifname.optional = true
 end
