diff --git a/package/network/utils/ipset/Makefile b/package/network/utils/ipset/Makefile
index bc4945e0f6..3d9d3343ab 100644
--- a/package/network/utils/ipset/Makefile
+++ b/package/network/utils/ipset/Makefile
@@ -59,9 +59,17 @@ define Build/InstallDev
 	$(CP) $(PKG_BUILD_DIR)/lib/libipset.pc $(1)/usr/lib/pkgconfig/
 endef
 
+define Package/ipset/conffiles
+/etc/ipset.conf
+endef
+
 define Package/ipset/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(CP) $(PKG_INSTALL_DIR)/usr/sbin/ipset $(1)/usr/sbin/
+	$(INSTALL_DIR) $(1)/etc/init.d
+	$(INSTALL_BIN) ./files/ipset.init $(1)/etc/init.d/ipset
+	$(INSTALL_DIR) $(1)/etc/
+	$(INSTALL_CONF) ./files/ipset.conf $(1)/etc/ipset.conf
 endef
 
 define Package/libipset/install
diff --git a/package/network/utils/ipset/files/ipset.conf b/package/network/utils/ipset/files/ipset.conf
new file mode 100644
index 0000000000..e69de29bb2
diff --git a/package/network/utils/ipset/files/ipset.init b/package/network/utils/ipset/files/ipset.init
new file mode 100755
index 0000000000..93e4b233ba
--- /dev/null
+++ b/package/network/utils/ipset/files/ipset.init
@@ -0,0 +1,15 @@
+#!/bin/sh /etc/rc.common
+
+START=18
+
+start() {
+    /usr/sbin/ipset -f /etc/ipset.conf restore
+}
+
+reload() {
+    /usr/sbin/ipset -f /etc/ipset.conf restore
+}
+
+stop() {
+    /usr/sbin/ipset destroy
+}
